examples/prelude/cpp/conan/BUILD

```diff
@@ -0,0 +1,54 @@
+load(
+    "@prelude//toolchains/conan:defs.bzl",
+    "conan_generate",
+    "conan_lock",
+    "conan_update",
+    "lock_generate",
+)
+load("//test_utils.bzl", "assert_output")
+
+conan_lock(
+    name = "lock",
+    conanfile = "conanfile.txt",
+    visibility = ["//cpp/conan/import:"],
+)
+
+lock_generate(
+    name = "lock-generate",
+    lockfile = ":lock",
+)
+
+# TODO[AH] Prevent double build of packages.
```

Comment:
This is interesting. I take it this cost only appears when actually generating the lockfile itself? That _seems_ like an ok trade-off for now.

Reply:
Correct, well, when generating the targets file, i.e. when invoking `conan install --generator ...`. But, yes, that's the idea.